lib/api-group.js

```diff
@@ -46,19 +46,31 @@ class ApiGroup {
     });
 
     //
-    // Create resources that live at the root (e.g., /api/v1/nodes)
+    // Create "group" resources that live at the root (e.g., /api/v1/nodes)
     //
-    for (const type of options.genericTypes) {
-      this[type] = new BaseObject({
-        name: type,
-        parentPath: this.path,
-        api: this
-      });
-    }
+    options.groupTypes.forEach(type => this.addResource(type));
+
     aliasResources(this);
   }
 
   /**
+   * Add a resource (e.g., nodes) to the ApiGroup group
+   * @param {string|object} options - resource name or options object
+   * @param {string} options.name - resource name
+   * @param {fn} options.Constructor - constructor for new resource
+   */
+  addResource(options) {
```

Comment:
Should we validate options to ensure it includes `name` and `Constructor` if it is an object?